src/Plugin.php

```diff
@@ -11,22 +11,222 @@
 namespace Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer;
 
 use Composer\Composer;
+use Composer\EventDispatcher\EventSubscriberInterface;
 use Composer\IO\IOInterface;
+use Composer\Package\AliasPackage;
+use Composer\Package\PackageInterface;
 use Composer\Plugin\PluginInterface;
+use Composer\Script\ScriptEvents;
+use Symfony\Component\Process\Exception\LogicException;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Exception\RuntimeException;
+use Symfony\Component\Process\Process;
 
 /**
  * PHP_CodeSniffer standard installation manager.
  *
  * @author Franck Nijhof <f.nijhof@dealerdirect.nl>
  */
-class Plugin implements PluginInterface
+class Plugin implements PluginInterface, EventSubscriberInterface
 {
+
+    const PACKAGE_TYPE = 'phpcodesniffer-standard';
+
+    /**
+     * @var Composer
+     */
+    protected $composer;
+
+    /**
+     * @var IOInterface
+     */
+    protected $io;
+
+    /**
+     * @var array
+     */
+    private $installedPaths;
+
+    /**
+     * @var string
+     */
+    private $phpCodeSnifferBin;
+
     /**
      * {@inheritDoc}
+     *
+     * @throws \RuntimeException
+     * @throws LogicException
+     * @throws RuntimeException
+     * @throws ProcessFailedException
      */
     public function activate(Composer $composer, IOInterface $io)
     {
-        $installer = new Installer($io, $composer);
-        $composer->getInstallationManager()->addInstaller($installer);
+        $this->composer = $composer;
+        $this->io = $io;
+        $this->installedPaths = [];
+        $this->phpCodeSnifferBin = $composer->getConfig()->get('bin-dir') . DIRECTORY_SEPARATOR . 'phpcs';
+
+        $this->loadInstalledPaths();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public static function getSubscribedEvents()
+    {
+        return [
+            ScriptEvents::POST_INSTALL_CMD => [
+                ['onScriptPost', 0],
+            ],
+            ScriptEvents::POST_UPDATE_CMD => [
+                ['onScriptPost', 0],
+            ],
+        ];
+    }
+
+    /**
+     * Entry point for post install and post update events
+     *
+     * @throws RuntimeException
+     * @throws LogicException
+     * @throws ProcessFailedException
+     */
+    public function onScriptPost()
+    {
+        // Ensure PHP_CodeSniffer is installed
+        if ($this->isPHPCodeSnifferInstalled() === false) {
+            return;
```

Comment:
False language construct. It would be better to invert the check (`!== false` or `=== true`) and move the statement below in to the `if` body. It might even be possible to collapse both `if`'s into one.